### PR TITLE
Add space padding to both sides of a return declaration colon

### PR DESCRIPTION
**Pull request by Crell**
As discussed on list.

### CODE REVIEW

**Comment by samdark** [comment]
That's subject for survey/voting. It's a single space after for many implementations so I don't think it's good to change it: http://www.typescriptlang.org/Handbook#interfaces

**Comment by michaelcullum** [comment]
:+1:

**Comment by samdark** [comment]
@michaelcullum how about conter-example of TypeScript?

File: proposed/extended-coding-style-guide.md [hunk 1]
```diff
@@ -479,8 +479,8 @@ $app->get('/hello/{name}', function ($name) use ($app) {
 });
 ```
 
-When you have a return type declaration present there MUST be one space after
-the colon with followed by the type declaration. The colon and declaration MUST be
+When you have a return type declaration present there MUST be one space preceeding and following
+the colon followed by the type declaration. The colon and declaration MUST be
```
Comment (donquixote) [inline proposed/extended-coding-style-guide.md]: typo? Should be "preceding".

Comment (samdark) [inline proposed/extended-coding-style-guide.md]: Yes.

**Comment by glen-84** [comment]
My vote: -1

^ Did you see what I did there?

Jokes aside – spaces on either side dissociates the type from the function declaration.

**Comment by PowerKiKi** [comment]
+1 should be merged

**Comment by michaelcullum** [comment]
Can we stop 'voting' please? This is waiting for the proper survey, and any +1s/-1s now will just be discarded.

**Comment by samdark** [comment]
Here's the survey for voicing your opinion. It would matter: http://goo.gl/forms/LyXppn4i5e

**Comment by KorvinSzanto** [comment]
@samdark I think it's probably not a good idea to distribute individual surveys like that. I've set aside some time next weekend to give that survey a pass, lets just focus on making the one monolith survey as complete as we can and release that.

**Comment by samdark** [comment]
What if more topics would arise?

**Comment by PowerKiKi** [comment]
I agree with @KorvinSzanto, a single survey as complete as possible would be for the best IMHO

**Comment by michaelcullum** [comment]
:+1: Korvin

**Comment by KorvinSzanto** [comment]
@samdark I'm definitely willing to discuss it as a group if you want, we can schedule a time in IRC to meet up or something and we can talk about whether we want to do individual surveys or a single huge one.

I figure if more issues come up after the survey, we can simply do another survey.

**Comment by samdark** [comment]
OK.

**Comment by hrvoj3e** [comment]
A single BIG survey would take too much time for people who track PSR-12 regularly 
_(Assuming it would have all questions I already answered before)._
It would make sense for people which are new to PSR-12 to express their opinion in bulk without needing to search for links to surveys in comments.

Could we have a thread or wiki page with links to all surveys and/or one big survey?

**Comment by samdark** [comment]
No, there's no big survey yet. It's in the making.

**Comment by PowerKiKi** [comment]
In case people here missed it, the survey is now available: https://goo.gl/forms/SonvjwqicPoBBwwl1

**Comment by pierniq** [comment]
from typographic point of view, in most non-programming languages (except french), there should be [no space before colon](https://en.wikipedia.org/wiki/Colon_(punctuation)#Spacing).

**Comment by donquixote** [comment]
@pierniq: I would argue that this colon is semantically different from a
non-programming language colon.

E.g. in a switch statement, the colon after an option is semantically
similar to normal language colon.
But in this case I think it is just an arbitrary symbol, and looking at it
from a normal-language perspective can be misleading.

This is not really meant as an argument for or against, I just want to
avoid that we do things for the wrong reasons.

**Comment by pierniq** [comment]
from PHP 7.1 nullable types was introduced.
it makes me sure, that "that" colon means the same as "natural language colon".
```php
function foo(): ?int {}
```
it means: "return value type could be:
- int,
- null."

**Comment by PowerKiKi** [comment]
I fail to see how the introduction of nullable types changes anything to the semantic. But more importantly than that, this issue was covered in the survey. And the survey result shall give this debate a final conclusion, [hopefully this week](https://groups.google.com/d/msg/php-fig/fGDuNFKy390/qBT_0XQpCQAJ). Meanwhile I don't think there is anything else constructive to add here.

**Comment by michaelcullum** [comment]
The survey was in favour of the current implementation.

**Pull request closed without merging**